src/vs/code/electron-main/window.ts

```diff
@@ -426,6 +430,11 @@ export class VSCodeWindow {
 		delete configuration.filesToCreate;
 		delete configuration.filesToDiff;
 
+		// Update untitled files to restore so they come through in the reloaded window
+		configuration.untitledToRestore = this.backupService.getWorkspaceUntitledFileBackupsSync(Uri.file(configuration.workspacePath)).map(filePath => {
```

Comment:
Fixed the general issue by removing redundant `loadSync` calls https://github.com/Microsoft/vscode/pull/14115/commits/209e0887c4ed5ae9db3d0c9aa43a3756587018b3

From what I know all logic up to this point is done synchronously so it needs to be sync at some point. Also this one is synchronous because it calls `readdirSync` to check if there are untitled backup files in the directory. This could be put in constructor where `loadSync` was moved in https://github.com/Microsoft/vscode/pull/14115/commits/209e0887c4ed5ae9db3d0c9aa43a3756587018b3 but that would only introduce an extra variable and still require sync IO.